### PR TITLE
ci: merge queue testing implementation

### PR DESCRIPTION
**Pull request by zack-is-cool**
addresses https://github.com/defenseunicorns/delivery-aws-iac/blob/main/doc/adr/0010-e2e-testing-improvement.md

I might move `.github/workflows/pr-merge-group.yml` to a reusable workflow instead of it living here, since it does have so much logic baked in

### CODE REVIEW

File: .github/workflows/pr-merge-group.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          region: us-east-2
+          github-context: "test / e2e-commercial-insecure"
+          test-to-run: "insecure"
+
+  # Only Run the E2E secure tests in the govcloud account while in the merge queue when it is a release-please branch into main.
+  # There is no direct way to determine if the  merge queued job is a release-please branch, so we check if this string is at the beginning of the commit message.
+  e2e-govcloud-secure:
+    if: |
+      github.event_name == 'merge_group' &&
+      startsWith(github.event.merge_group.head_commit.message, 'chore(main): release')
```
Comment (ntwkninja) [inline .github/workflows/pr-merge-group.yml:78]: This feels brittle. Could we trigger based on source branch `release-please--branches--main` & target branch `main`

Comment (zack-is-cool) [reply .github/workflows/pr-merge-group.yml:78]: Unfortunately that isn't available in the merge_group events. It does the merge from a new temporary branch whose name has the source PR number in it. 

I could try to do some extra logic to query github and check the source PR, I'll just need to trim the PR # out and build a URL to try to fetch the head branch of the PR and pull it out of that response

Comment (zack-is-cool) [reply .github/workflows/pr-merge-group.yml:78]: https://github.com/defenseunicorns/delivery-aws-iac/actions/runs/6240220509/job/16939832808

Check out this debug job here for what is in a `merge_group` event

**Review by ntwkninja — DISMISSED** [review]
lgtm

**Review by ntwkninja — APPROVED** [review]
🚀

**Pull request merged**